"lint" (`golangci-lint`) sometimes misses !

our lint has some issues, it acts differently on different computers and sometimes misses things like this : 
https://github.com/kubernetes/minikube/pull/6712

Fails because of missing libvirt, probably have to make our own downstream image:

`level=warning msg="[runner] Can't run linter unused: buildssa: analysis skipped: errors in package: [/app/pkg/drivers/kvm/domain.go:28:10: could not import github.com/libvirt/libvirt-go (/go/pkg/mod/github.com/libvirt/libvirt-go@v3.4.0+incompatible/callbacks.go:63:8: could not import C (cgo preprocessing failed))]"
`

--- a/Makefile
+++ b/Makefile
@@ -364,11 +364,12 @@ out/linters/golangci-lint-$(GOLINT_VERSION):
 
 # this one is meant for local use
 .PHONY: lint
-lint: pkg/minikube/assets/assets.go pkg/minikube/translate/translations.go out/linters/golangci-lint-$(GOLINT_VERSION) ## Run lint
 ifeq ($(MINIKUBE_BUILD_IN_DOCKER),y)
+lint: pkg/minikube/assets/assets.go pkg/minikube/translate/translations.go
 	docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:$(GOLINT_VERSION) \
 	golangci-lint run ${GOLINT_OPTIONS} --skip-dirs "cmd/drivers/kvm|cmd/drivers/hyperkit|pkg/drivers/kvm|pkg/drivers/hyperkit" ./...
 else
+lint: pkg/minikube/assets/assets.go pkg/minikube/translate/translations.go out/linters/golangci-lint-$(GOLINT_VERSION) ## Run lint
 	./out/linters/golangci-lint-$(GOLINT_VERSION) run ${GOLINT_OPTIONS} ./...
 endif
 

--- a/Makefile
+++ b/Makefile
@@ -365,7 +365,11 @@ out/linters/golangci-lint-$(GOLINT_VERSION):
 # this one is meant for local use
 .PHONY: lint
 lint: pkg/minikube/assets/assets.go pkg/minikube/translate/translations.go out/linters/golangci-lint-$(GOLINT_VERSION) ## Run lint
+ifeq ($(MINIKUBE_BUILD_IN_DOCKER),y)
+	docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:$(GOLINT_VERSION) golangci-lint run ${GOLINT_OPTIONS} ./...
+else
 	./out/linters/golangci-lint-$(GOLINT_VERSION) run ${GOLINT_OPTIONS} ./...
+endif
 
 # lint-ci is slower version of lint and is meant to be used in ci (travis) to avoid out of memory leaks.
 .PHONY: lint-ci

--- a/Makefile
+++ b/Makefile
@@ -366,7 +366,8 @@ out/linters/golangci-lint-$(GOLINT_VERSION):
 .PHONY: lint
 lint: pkg/minikube/assets/assets.go pkg/minikube/translate/translations.go out/linters/golangci-lint-$(GOLINT_VERSION) ## Run lint
 ifeq ($(MINIKUBE_BUILD_IN_DOCKER),y)
-	docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:$(GOLINT_VERSION) golangci-lint run ${GOLINT_OPTIONS} ./...
+	docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint:$(GOLINT_VERSION) \
+	golangci-lint run ${GOLINT_OPTIONS} --skip-dirs "cmd/drivers/kvm|cmd/drivers/hyperkit|pkg/drivers/kvm|pkg/drivers/hyperkit" ./...
 else
 	./out/linters/golangci-lint-$(GOLINT_VERSION) run ${GOLINT_OPTIONS} ./...
 endif

